src/test/java/jenkins/branch/OrganizationFolderTest.java

```diff
@@ -140,6 +149,171 @@ public void deletedMarker() throws Exception {
+                    org_factory_is_default_factory = true;
+                    break;
+                }
+            }
+
+            assertTrue(
+                "After #onCreatedFromScratch() with no initial project factories, the OrganizationFolder MultiBranchProjectFactory [" + org_factory.getDescriptor().getDisplayName()+ "] is one of the enabled-by-default factories.",
+                org_factory_is_default_factory );
+        }
+    }
+
+    /**
+     * When an OrganizationFolder is created and provided with at least one {@link MultiBranchProjectFactory},
+     * it should not add any other factories when {@link OrganizationFolder#onCreatedFromScratch()} is called.
+     */
+    @Test
+    public void noDefaultFactoriesWhenNotNeeded() {
+
+        // programmatically create an OrganizationFolder
+        OrganizationFolder newbie = new OrganizationFolder( r.jenkins, "newbie" );
+
+        // it should have no factories
+        assertEquals(
+            "A newly-created OrganizationFolder has no MultiBranchProjectFactories.",
+            0,
+            newbie.getProjectFactories().size() );
+
+        // set exactly one non-default factory
+        newbie.getProjectFactories().clear();
+        MockFactory mock_factory = new MockFactory();
+        newbie.getProjectFactories().add( mock_factory );
+
+        // at some point after creating it, onCreatedFromScratch() would be called by Jenkins
+        newbie.onCreatedFromScratch();
+
+        assertEquals(
+            "An OrganizationFolder created with a non-default project factory should only have one factory after onCreatedFromScratch()",
+            1,
+            newbie.getProjectFactories().size() );
+
+        assertEquals(
+            "An OrganizationFolder created with a non-default project factory should only have that particular factory after onCreatedFromScratch()",
+            mock_factory.getDescriptor(),
+            newbie.getProjectFactories().get( 0 ).getDescriptor() ) ;
+
+    }
+
+    @Test
+    public void modifyAclsWhenInComputedFolder() throws IOException, InterruptedException {
+
+        Set<Permission> suppressed_permissions =
+            ImmutableSet.of(Item.CONFIGURE, Item.DELETE, View.CONFIGURE, View.CREATE, View.DELETE);
+
+        // we need a ComputedFolder to be the parent of our OrganizationFolder
+        ComputedFolder<OrganizationFolder> computed_folder = new ComputedFolder<OrganizationFolder>(r.jenkins, "top") {
+
+            @Override
+            protected void computeChildren(
+                ChildObserver<OrganizationFolder> observer,
+                TaskListener listener)
+                    throws IOException, InterruptedException
+            {
+                /*
+                 * doesn't look like we can actually compute children during a unit test.
+                 * Normally we'd want to create the OrganizationFolder here
+                 * to match the "real" Jenkins workflow as closely as possible
+                 */
```

Comment:
@awittha 
Why can't we compute children during a unit test?

Reply:
In my own code, when I want to kick off the computation of children of a ComputedFolder in a "safe" way, I used `ComputedFolder#scheduleBuild(...)`.

This didn't actually result in any computations happening, during the unit tests - the actual `computeChildren(...)` method was never called.

I didn't attach a debugger, but my _guess_  is that when it tries to actually schedule a build ( https://github.com/jenkinsci/cloudbees-folder-plugin/blob/master/src/main/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolder.java#L580 ), there isn't actually a build queue in the mocked-up Jenkins?

In any case, it isn't terribly important to _actually_ compute children as this code doesn't activate _when an `OrganizationFolder` is created as part of a `ComputedFolder` computation_, but merely when an `OrganizationFolder`'s _parent is a `ComputedFolder`_ - which we can set up without actually having to have a working `ComputedFolder` during the test.

Should I update the comment to be more specific to that effect - that it doesn't work out of the box and we don't really need it anyway - do you think?

Reply:
If you want to add a little more to say why it is not a big deal, that'd be great. 

Great work on this.

Reply:
Updated the comment; new text:

```
* We don't actually need to compute children during a unit test.
* We just need an OrganizationFolder to have this ComputedFolder
* as its parent.
*
* Since item parents are set on the item when constructing it,
* we can do that when we create the OrganizationFolder.
*
* Also, #scheduleBuild(...) to enqueue a computeChildren
* doesn't work out-of-the-box in the current unit-test setup.
```